### PR TITLE
add e2e tmpl test

### PR DESCRIPTION
**Pull request by yxxhero**
Signed-off-by: yxxhero <aiopsclub@163.com>

add e2e tmpl test

https://github.com/helmfile/helmfile/discussions/100

### CODE REVIEW

**Review by itscaro — APPROVED** [review]

**Comment by yxxhero** [comment]
@itscaro what do you think of?

**Comment by itscaro** [comment]
@mumoshu any comment?

**Comment by yxxhero** [comment]
@itscaro ok. I wiil add more comments. Thanks very much.

File: test/e2e/template/helmfile/tmpl_test.go [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+	t.append(tplTestCases...)
+}
+
+var tmplE2eTest = tmplE2e{}
+
+// TestTmplStrings tests the template string
+func TestTmplStrings(t *testing.T) {
+	c := &tmpl.Context{}
+	c.SetBasePath(".")
+	c.SetReadFile(os.ReadFile)
```
Comment (mumoshu) [inline test/e2e/template/helmfile/tmpl_test.go:231]: Nit but we'd better just change basePath and readFile fields to public (BasePath and ReadFile respectively) in this case. It should be more idiomatic-go when the function is going to be a simple "setter"!

File: test/e2e/template/helmfile/tmpl_test.go [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+	tmpl := template.New("stringTemplateTest").Funcs(c.CreateFuncMap())
+
+	tmplE2eTest.load()
```
Comment (mumoshu) [inline test/e2e/template/helmfile/tmpl_test.go:234]: Sorry for very nit-picky again but I personally prefer it being an inlined sequence of `tcs = append(tcs, requireEnvTestCases...)` without the indirection(extra struct and its instance plus another two functions to just compose an array of test cases).

**Review by mumoshu — APPROVED** [review]
A few nits but LGTM overall! Thanks for your continued effort ☺️

File: test/e2e/template/helmfile/tmpl_test.go (hunk 1 above)
Comment (yxxhero) [reply test/e2e/template/helmfile/tmpl_test.go:231]: I will continuous optimization. Thanks very much. @mumoshu

File: test/e2e/template/helmfile/tmpl_test.go (hunk 2 above)
Comment (yxxhero) [reply test/e2e/template/helmfile/tmpl_test.go:234]: I will continuous optimization. Thanks very much. @mumoshu

**Pull request merged**